src/libraries/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -639,13 +639,13 @@ private bool AcquireServerCredentials(ref byte[]? thumbPrint, ReadOnlySpan<byte>
             // with .NET Framework), and if neither is set we fall back to using ServerCertificate.
             if (_sslAuthenticationOptions.ServerCertSelectionDelegate != null)
             {
-                string? serverIdentity = SniHelper.GetServerName(clientHello);
-                localCertificate = _sslAuthenticationOptions.ServerCertSelectionDelegate(serverIdentity);
-
+                localCertificate = _sslAuthenticationOptions.ServerCertSelectionDelegate(_sslAuthenticationOptions.TargetHost);
                 if (localCertificate == null)
                 {
                     throw new AuthenticationException(SR.net_ssl_io_no_server_cert);
                 }
+                if (NetEventSource.IsEnabled)
+                    NetEventSource.Info(this, "Use delegate selected Cert");
```

Comment:
Worth including the cert in the log?  Do we do that elsewhere?
```suggestion
                    NetEventSource.Info(this, $"Using certificate returned by selection delegate: {localCertificate}");
```

Reply:
Certs don't ToString in a particularly log-friendly format.  If you want to include it, I'd recommend logging it as the subject name, or thumbprint, or whatever.

Reply:
I only added this because of same entry in block bellow. I agree with @bartonjs and I already started tracing scrub through SslStream. The outcome will come as separate PR.